apps/application-system/api/src/app/modules/application/application.controller.ts

```diff
@@ -1128,6 +1139,48 @@ export class ApplicationController {
         'Users role does not have permission to delete this application in this state',
       )
     }
+
+    // delete charge in FJS
+    this.deleteApplicationCharge(existingApplication)
+
+    // delete the entry in Payment table to prevent FK error
+    await this.paymentService.delete(existingApplication.id, user)
+
     await this.applicationService.delete(existingApplication.id)
   }
+
+  private async deleteApplicationCharge(application: BaseApplication) {
+    try {
```

Comment:
You should move this to a separate service within and so you can use it here and for the pruning process to avoid code duplication.